nemo/collections/tts/modules/aligner.py

```diff
@@ -57,15 +78,35 @@ def get_dist(self, keys, queries, mask=None):
         Output:
             dist (torch.tensor): B x T1 x T2 tensor.
         """
-        keys_enc = self.key_proj(keys)  # B x n_attn_dims x T2
-        queries_enc = self.query_proj(queries)  # B x n_attn_dims x T1
-        attn = (queries_enc[:, :, :, None] - keys_enc[:, :, None]) ** 2  # B x n_attn_dims x T1 x T2
-        dist = attn.sum(1, keepdim=True)  # B x 1 x T1 x T2
+        # B x n_attn_dims x T2
+        keys_enc = self.key_proj(keys)
+        # B x n_attn_dims x T1
+        queries_enc = self.query_proj(queries)
+
+        # B x 1 x T1 x T2
+        dist = self.dist_fn(queries=queries_enc, keys=keys_enc)
+
+        self._apply_mask(dist, mask, float("inf"))
 
-        if mask is not None:
-            dist.data.masked_fill_(mask.permute(0, 2, 1).unsqueeze(2), float("inf"))
+        return dist
 
-        return dist.squeeze(1)
+    @staticmethod
+    def get_euclidean_dist(queries, keys):
+        queries = rearrange(queries, "B C T1 -> B C T1 1")
```

Comment:
same comment as the above. Please choose either `C` or `n_attn_dims`, but not mix usage of them.